source/shared/components/App.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import settings from 'server/settings';
+import Title from './Title';
+
+const App = ({ dispatch, books }) => {
+  const bookNodes = books.items.map(book => {
+    return (
+      <div key={ book.id }>
+        { book.text } - Read by { book.count } people.
+        <button onClick={ () => dispatch({ type: 'ADD_COUNT', book }) }>Add reader</button>
```

Comment:
I wouldn't worry about this unless it is actually identified as a performance issue in the application. Maybe if you're making a video game and animating thousands of components on the screen at 60fps...